### PR TITLE
yaml-cpp: Fix visibility attribute for ubuntu

### PR DESCRIPTION
**Pull request by JafarAbdi**
**Describe the pull request**

I'm building the libraries as shared libraries as shown in [overlay-triplets-linux-dynamic](https://github.com/microsoft/vcpkg/blob/master/docs/examples/overlay-triplets-linux-dynamic.md) which cause the `dll.h` header in `yaml-cpp` package to always define `YAML_CPP_API` as `__declspec(dllexport)` or `__declspec(dllimport)` ([link](https://github.com/microsoft/vcpkg/blob/master/ports/yaml-cpp/portfile.cmake#L39-L43)) which is causing compile errors when built on gcc/clang in ubuntu, this PR replaces `__declspec(dllexport)`/`__declspec(dllimport)` with `__attribute__ ((visibility ("default")))` as shown in [GNU C++ visibility](https://gcc.gnu.org/wiki/Visibility#Step-by-step_guide)

- #### What does your PR fix?  
  Fixes #...

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  linux, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/yaml-cpp/portfile.cmake [hunk 1]
```diff
@@ -38,6 +38,10 @@ file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 file(READ ${CURRENT_PACKAGES_DIR}/include/yaml-cpp/dll.h DLL_H)
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
     string(REPLACE "#ifdef YAML_CPP_DLL" "#if 1" DLL_H "${DLL_H}")
+    if(NOT (WIN32 OR CYGWIN))
+        string(REPLACE "__declspec(dllexport)" "__attribute__ ((visibility (\"default\")))" DLL_H "${DLL_H}")
+        string(REPLACE "__declspec(dllimport)" "__attribute__ ((visibility (\"default\")))" DLL_H "${DLL_H}")
+    endif()
```
Comment (JackBoosY) [inline ports/yaml-cpp/portfile.cmake:44]: Are there other parts of the source code that contain `__attribute__` tags?
If not, it means that the port does not support non-windows upstream.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Ping @JafarAbdi for response.

**Comment by JafarAbdi** [comment]
> Ping @JafarAbdi for response.

Super sorry for the delay

File: ports/yaml-cpp/portfile.cmake (hunk 1 above)
Comment (JafarAbdi) [reply ports/yaml-cpp/portfile.cmake]: > Are there other parts of the source code that contain __attribute__ tags?

I think this's the only place that contains the `__attribute__` tag

> it means that the port does not support non-windows upstream.

The main library does support non-windows upstream, replacing `#ifdef YAML_CPP_DLL` with `#if 1` is what causes it to fail on Ubuntu for example

**Comment by JafarAbdi** [comment]
@JackBoosY I did update the condition to replace `YAML_CPP_DLL`, what do you think about this approach .?

Thanks!

File: versions/y-/yaml-cpp.json [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-      "git-tree": "fabfdd85c28d751aa769d2e743bf0e9ccd8dd178",
+      "git-tree": "9b7bab809af815c232a8bcf41af6d438c2563ba6",
       "version-string": "0.6.3",
       "port-version": 0
     },
+    {
+      "git-tree": "fabfdd85c28d751aa769d2e743bf0e9ccd8dd178",
+      "version-string": "0.6.3",
+      "port-version": 1
+    },
```
Comment (JackBoosY) [inline versions/y-/yaml-cpp.json:12]: Please revert these changes, bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).
Then run command `vcpkg x-add-version --all` then commit the changes.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

File: ports/yaml-cpp/portfile.cmake [hunk 3]
```diff
@@ -36,7 +36,7 @@ endif()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 file(READ ${CURRENT_PACKAGES_DIR}/include/yaml-cpp/dll.h DLL_H)
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic" AND (WIN32 OR CYGWIN))
```
Comment (strega-nil-ms) [inline ports/yaml-cpp/portfile.cmake:39]:
```suggestion
if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic" AND VCPKG_TARGET_IS_WINDOWS)
```

**Comment by strega-nil-ms** [comment]
@JafarAbdi could you make the requested change?

**Comment by JafarAbdi** [comment]
> @JafarAbdi could you make the requested change?

Yep, done

Thanks!

**Comment by JafarAbdi** [comment]
I had to rerun `vcpkg x-add-version --all` that's why I force-pushed the branch
